android/sdk/src/main/java/org/jitsi/meet/sdk/dropbox/Dropbox.java

```diff
@@ -0,0 +1,196 @@
+package org.jitsi.meet.sdk.dropbox;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxRequestConfig;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.users.FullAccount;
+import com.dropbox.core.v2.users.SpaceAllocation;
+import com.dropbox.core.v2.users.SpaceUsage;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.dropbox.core.android.Auth;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.WritableMap;
+import org.jitsi.meet.sdk.R;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Implements the react-native module for the dropbox integration.
+ */
+public class Dropbox extends ReactContextBaseJavaModule implements LifecycleEventListener {
+
+    private Promise promise = null;
+    private String clientId;
+    private boolean isEnabled = false;
+
+    public Dropbox(ReactApplicationContext reactContext) {
+        super(reactContext);
+        reactContext.addLifecycleEventListener(this);
+        clientId = generateClientId();
+        String appID = this.getCurrentActivity().getString(R.string.dropbox_app_key);
```

Comment:
How secret does this key need to be? Can it be placed in config.js and passed along with a method call?

Reply:
This code will be changed a little. Check out @paweldomas 's comment.

But generally I wasn't been able to reuse config.js 's client id. Otherwise this would be the perfect solution, so if you have any idea how to achieve it please share it!

Reply:
How were you not able to do it? Does it not work? All we need is a method which can be called from JS which passes the appId, which is fetched from config.js... or am I missing something?

Reply:
You can see that we need to put client Id into the manifest file as well, as part of the activity that starts the OAuth flow. Maybe you missed that or maybe there is a workaround for this problem that I'm not aware of?